File: sandbox/airavata-kubernetes/modules/microservices/api-server/src/main/java/org/apache/airavata/k8s/api/server/model/compute/ComputeResourceModel.java
/**
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.airavata.k8s.api.server.model.compute;

import javax.persistence.*;

/**
 * TODO: Class level comments please
 *
 * @author dimuthu
 * @since 1.0.0-SNAPSHOT
 */
@Entity
@Table(name = "COMPUTE_RESOURCE")
public class ComputeResourceModel {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    private String name;
    private String host;
    private String userName;
    private String password;
    private String communicationType;


    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getHost() {
        return host;
    }

    public ComputeResourceModel setHost(String host) {
        this.host = host;
        return this;
    }

    public String getUserName() {
        return userName;
    }

    public ComputeResourceModel setUserName(String userName) {
        this.userName = userName;
        return this;
    }

    public String getPassword() {
        return password;
    }

    public ComputeResourceModel setPassword(String password) {
        this.password = password;
        return this;
    }

    public String getCommunicationType() {
        return communicationType;
    }

    public ComputeResourceModel setCommunicationType(String communicationType) {
        this.communicationType = communicationType;
        return this;
    }
}

File: sandbox/airavata-kubernetes/web-console/src/app/models/compute/compute.resource.model.ts
/**
 * TODO: Class level comments please
 *
 * @author dimuthu
 * @since 1.0.0-SNAPSHOT
 */
export class ComputeResource {

  id: number;
  name: string;
  host: string;
  userName: string;
  password: string;
  communicationType: string;

  constructor(id: number = 0, name: string = null, host: string = null, userName: string = null,
              password: string = null, communicationType: string = "Mock") {
    this.id = id;
    this.name = name;
    this.host = host;
    this.userName = userName;
    this.password = password;
    this.communicationType = communicationType;
  }

  public static fromJson(json:any): ComputeResource {
    return new ComputeResource(json.id, json.name, json.host, json.userName,
      json.password, json.communicationType);
  }
}
